core/src/com/unciv/logic/automation/NextTurnAutomation.kt

```diff
@@ -300,6 +301,39 @@ object NextTurnAutomation {
         }
     }
 
+    /** If we are able to build a spaceship but have already spent our resources, try disbanding
+     *  a unit and selling a building to make room. Can happen due to trades etc */
+    private fun freeUpSpaceResources(civInfo: CivilizationInfo) {
+        // Can't build spaceships
+        if (!civInfo.hasUnique("Enables construction of Spaceship parts"))
+            return
+
+        for (resource in civInfo.gameInfo.spaceResources) {
+            // Have enough resources already
+            if (civInfo.getCivResourcesByName()[resource]!! >= Automation.getReservedSpaceResourceAmount(civInfo))
+                continue
+
+            val unitToDisband = civInfo.getCivUnits()
+                .filter { it.baseUnit.getResourceRequirements().containsKey(resource) }
+                .toList().minByOrNull { it.getForceEvaluation() }
```

Comment:
Why tolist?

Reply:
Because you can't minby a sequence?

Reply:
Can't you? I recall doing so

Reply:
sequences are almighty! Of course you can aggregate any way you want - you can even asSequence() them!
<details><summary>implementation</summary>

```kotlin
/**
 * Returns the first element yielding the smallest value of the given function or `null` if there are no elements.
 *
 * The operation is _terminal_.
 * 
 * @sample samples.collections.Collections.Aggregates.minByOrNull
 */
@SinceKotlin("1.4")
public inline fun <T, R : Comparable<R>> Sequence<T>.minByOrNull(selector: (T) -> R): T? {
    val iterator = iterator()
    if (!iterator.hasNext()) return null
    var minElem = iterator.next()
    if (!iterator.hasNext()) return minElem
    var minValue = selector(minElem)
    do {
        val e = iterator.next()
        val v = selector(e)
        if (minValue > v) {
            minElem = e
            minValue = v
        }
    } while (iterator.hasNext())
    return minElem
}
```
</details>

Reply:
don't know why I thought that then!